meta-ostro/classes/image-dsk.bbclass

```diff
@@ -24,16 +24,91 @@
 #   This is required to identify it and pass its Partition UUID to the kernel, for booting.
 
 
-COMPRESSIONTYPES_append = " vdi"
+COMPRESSIONTYPES_append = " vdi ova"
 COMPRESS_CMD_vdi = "qemu-img convert -O vdi ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type} ${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${type}.vdi"
+
 COMPRESS_DEPENDS_vdi = "qemu-native"
+COMPRESS_DEPENDS_ova = "vboxmanage-native"
 
 IMAGE_DSK_ACTIVE = "${@ bool([x for x in d.getVar('IMAGE_FSTYPES', True).split() if x == 'dsk' or x.startswith('dsk.')])}"
+
+
 python () {
     if d.getVar('IMAGE_DSK_ACTIVE', True) == 'True':
         d.setVar('IMAGE_NAME_SUFFIX', '')
 }
 
+
+create_ova() {
+
+  if [ "x${MACHINE}x" != "xintel-corei7-64x" ]; then
+    exit 0
+  fi
+
+  # 64 bit linux with sata bus
+  OS_TYPE="Linux_64"
+  STORAGE_NAME="SATA"
+  STORAGE_BUS_TYPE="sata"
+
+
+  VM_NAME="${IMAGE_NAME}"
+  # 512 should be large enough value to run Ostro (Galileo has 256MB RAM for example)
+  # but small enough value that even a modest host machine can afford it
+  RAM="512"
+  # min allowed - as we do not run graphical environment, it should be enough
+  VRAM="12"
+  # As above with RAM, 2 cores should be enough for Ostro without crippling the host machine
+  CPU_CORE_COUNT=2
+  # required for successful boot
+  FIRMWARE="efi"
+  IOAPIC="on"
+
+  # default settings for hard drive
+  STORAGE_PORT="0"
+  STORAGE_DEVICE="0"
+  STORAGE_DEVICE_TYPE="hdd"
+
+  VIRTUALBOX_IMAGE="${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.dsk.vdi"
+  VIRTUALBOX_EXECUTABLE="${STAGING_BINDIR_NATIVE}/VBoxManage"
+
+  APPLIANCE_NAME="${VM_NAME}.dsk.ova"
+
+  FOLDER_NAME="virtualbox-vm"
+  WORK_FOLDER="${DEPLOY_DIR_IMAGE}/${FOLDER_NAME}"
+
+  mkdir -p ${WORK_FOLDER}
+
+  # erase any temp files from previous builds, as these can cause issues
+  # notably NS_ERROR_FACTORY_NOT_REGISTERED (0x80040154)
+  sudo find /tmp -name ".vbox-root*" -type d -prune -exec rm -rf {} \;
```

Comment:
sudo isn't right. this can be modified such that you only search for files in /tmp/.vbox-error (I'm assuming you wanted to get of the 'permission denied' when scanning /tmp.)